### PR TITLE
schematelemetry: test that logs are redacted correctly

### PR DESCRIPTION
**Pull request by rafiss**
follow-up from https://github.com/cockroachdb/cockroach/pull/127195
informs: https://github.com/cockroachdb/cockroach/issues/126584
Release note: None

### CODE REVIEW

File: pkg/sql/catalog/schematelemetry/schema_telemetry_test.go [hunk 1]
```diff
@@ -190,4 +190,11 @@ UPDATE system.namespace SET id = %d WHERE id = %d;
 	entries, err := log.FetchEntriesFromFiles(0, math.MaxInt64, 1000, errorRE, log.SelectEditMode(false, false))
```
Comment (annrpom) [inline pkg/sql/catalog/schematelemetry/schema_telemetry_test.go:190]: supernit: could we also annotate the arguments to SelectEditMode here

File: pkg/sql/catalog/schematelemetry/schema_telemetry_test.go [hunk 2]
```diff
@@ -190,4 +190,11 @@ UPDATE system.namespace SET id = %d WHERE id = %d;
...
 	require.NoError(t, err)
 	require.Len(t, entries, 9)
+
+	// Verify that the log entries have redaction markers applied by checking one
+	// of the specific error messages.
+	errorRE = regexp.MustCompile(`found invalid object with ID \d+: relation ‹"nojob"›`)
+	entries, err = log.FetchEntriesFromFiles(0, math.MaxInt64, 1000, errorRE, log.SelectEditMode(false, true /* keepRedactable */))
```
Comment (annrpom) [inline pkg/sql/catalog/schematelemetry/schema_telemetry_test.go:197]: supernit: could we also annotate the first argument to SelectEditMode with /* redact */

**Review by annrpom — APPROVED** [review]
:lgtm:

File: pkg/sql/catalog/schematelemetry/schema_telemetry_test.go (hunk 1 above)
Comment (rafiss) [reply pkg/sql/catalog/schematelemetry/schema_telemetry_test.go]: done

File: pkg/sql/catalog/schematelemetry/schema_telemetry_test.go (hunk 2 above)
Comment (rafiss) [reply pkg/sql/catalog/schematelemetry/schema_telemetry_test.go]: done

**Comment by rafiss** [comment]
tftr! 

bors r+

**Pull request merged**